tools/serverpod_cli/lib/src/analyzer/models/definitions.dart

```diff
@@ -253,6 +261,35 @@ class ProtocolEnumValueDefinition {
   ProtocolEnumValueDefinition(this.name, [this.documentation]);
 }
 
+abstract class InheritanceDefinition {
+  final String className;
+
+  InheritanceDefinition(this.className);
+}
+
+class UnresolvedInheritanceDefinition extends InheritanceDefinition {
+  UnresolvedInheritanceDefinition(super.className);
+}
+
+class BaseClassInheritanceDefinition extends InheritanceDefinition {
+  final List<ClassDefinition?> subClasses;
+
+  BaseClassInheritanceDefinition(
+    super.className,
+    this.subClasses,
+  );
+}
```

Comment:
Suggestion/Discussion: Rename to `ParentClass`/`ChildClass` or `SuperClass`/`SubClass` - `InheritanceDefinition`.

Since "Base" actually has a meaning in a class [context in Dart](https://dart.dev/language/class-modifiers#base), this naming can be a bit confusing.
For me, it is also a bit unclear what the relationship is to the class from the perspective of this class.

By using the term `Parent`/`Child` or `Super`/`Sub` class I think this becomes a bit more clear.